src/Metrics/Reporter.h

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#pragma once
+
+#include "InternalMetric.h"
+#include "Sink.h"
+#include <asio.hpp>
+#include <map>
+#include <memory>
+#include <thread>
+
+namespace Metrics {
+
+class Reporter {
+public:
+  Reporter(std::unique_ptr<Sink> MetricSink, std::chrono::milliseconds Interval)
+      : MetricSink(std::move(MetricSink)), IO(), Period(Interval),
+        AsioTimer(IO, Period){};
+
+  virtual ~Reporter() = default;
```

Comment:
We should print an error if there is one or more metric that has not been de-registered when calling the destructor.

Reply:
Do we care if `Reporter` doesn't outlive all `Metric`s

Reply:
If this happens, then it is an indication that there likely is a serious bug in the code.